src/slider/internal.tsx

```diff
@@ -61,19 +82,24 @@ export default function InternalSlider({
     warnOnce('Slider', 'The step value cannot be greater than the difference between the min and max.');
   }
 
-  const tooltip = hasValue && <Tooltip value={valueFormatter ? valueFormatter(value) : value} trackRef={handleRef} />;
+  if (step && hasValue && (value - min) % step !== 0) {
+    warnOnce('Slider', 'Slider value must be a multiple of the step. The value will round to the nearest step value.');
+  }
+
+  const tooltip = <Tooltip value={valueFormatter ? valueFormatter(getValue()) : getValue()} trackRef={handleRef} />;
 
   const getAriaValueText = () => {
-    if (value !== undefined && valueFormatter && valueFormatter(value)) {
-      return valueFormatter(value);
+    if (valueFormatter && valueFormatter(getValue())) {
```

Comment:
Instead of calling the getValue() multiple times can we compute the normalised value once and then use it?